cmake/iotjs.cmake

```diff
@@ -25,27 +25,21 @@ function(find_value RESULT VALUE VALUE_TRUE VALUE_FALSE)
   endif()
 endfunction(find_value)
 
-# System Configuration
+# System Configuration (not module)
 set(IOTJS_PLATFORM_SRC)
-if("${CMAKE_SYSTEM_NAME}" MATCHES "Linux|Tizen")
-  file(GLOB IOTJS_PLATFORM_LINUX_SRC
-       ${IOTJS_SOURCE_DIR}/platform/iotjs_*-linux.c)
-  list(APPEND IOTJS_PLATFORM_SRC ${IOTJS_PLATFORM_LINUX_SRC})
-  # in case of Tizen force the platform to Linux
-  string(REGEX REPLACE "tizen$" "linux"
-         PLATFORM_DESCRIPTOR "${PLATFORM_DESCRIPTOR}")
-endif()
+string(TOLOWER ${CMAKE_SYSTEM_NAME} IOTJS_SYSTEM_OS)
+set(PLATFORM_OS_SRC
+    ${IOTJS_SOURCE_DIR}/platform/${IOTJS_SYSTEM_OS})
+set(OS_SRC ${PLATFORM_OS_SRC}/iotjs_[^module]*${IOTJS_SYSTEM_OS}.c)
+file(GLOB IOTJS_OS_SRC ${OS_SRC})
+list(APPEND IOTJS_PLATFORM_SRC ${IOTJS_OS_SRC})
 
 # Board Configuration (not module)
 if(DEFINED TARGET_BOARD)
-  string(TOLOWER "${TARGET_BOARD}" BOARD_DESCRIPTOR)
-
-  if("${BOARD_DESCRIPTOR}" STREQUAL "stm32f4dis")
-    set(BOARD_DESCRIPTOR stm32)
-  endif()
-
-  set(BOARD_SRC ${IOTJS_SOURCE_DIR}/platform/${PLATFORM_DESCRIPTOR})
-  set(BOARD_SRC ${BOARD_SRC}/iotjs_[^module]*${BOARD_DESCRIPTOR}.c)
+  set(PLATFORM_BOARD_SRC ${IOTJS_SOURCE_DIR}/platform/${IOTJS_SYSTEM_OS})
+  set(PLATFORM_BOARD_SRC ${PLATFORM_BOARD_SRC}/${IOTJS_SYSTEM_OS})
+  set(PLATFORM_BOARD_SRC ${PLATFORM_BOARD_SRC}-${TARGET_BOARD})
+  set(BOARD_SRC ${PLATFORM_BOARD_SRC}/iotjs_[^module]*${IOTJS_SYSTEM_OS}.c)
```

Comment:
This is too many steps, again (and/or misleading variable names), IMHO.
```cmake
set(IOTJS_BOARD_DIR ${IOTJS_PLATFORM_DIR}/${IOTJS_SYSTEM_OS}-${TARGET_BOARD})
file(GLOB IOTJS_BOARD_SRC ${IOTJS_BOARD_DIR}/iotjs_[^module]*-${IOTJS_SYSTEM_OS}.c)
```

BTW, I'm not sure that the repetition in `nuttx/nuttx-stm32f4dis` is necessary, perhaps a directory structure of `nuttx/stm32f4dis` could be just as expressive. I'll raise this remark on the issue as well but mention it here, too, as it affects the required glob pattern.